radial_build_tools.py

```diff
@@ -56,6 +54,38 @@ def __init__(self, build, **kwargs):
         ):
             self.__setattr__(name, kwargs[name])
 
+        self.used_colors = set()
+        self.available_colors = set(matplotlib.colors.XKCD_COLORS.values())
+        self.colors = self.assign_colors()
+
+    def assign_colors(self):
+        """
+        Assign colors to each layer. If a color is provided, use it.
+        If not, generate a random color that hasn't been used yet.
+        """
+        colors = []
+        for layer in self.build.values():
+            if "color" in layer:
+                color = layer["color"]
+                self.used_colors.add(color)
+                self.available_colors.discard(color)
+            else:
+                color = self.generate_unique_color()
+                layer["color"] = color  
+
+            colors.append(color)
+
+        return colors
+
+    def generate_unique_color(self):
+        """
+        Generate a random color that has not been used yet.
+        """
+        color = random.choice(list(self.available_colors))
+        self.available_colors.remove(color)  # Remove chosen color
+        self.used_colors.add(color)  # Add to used colors
+        return color
```

Comment:
If you streamline where you add/remove colors from sets, you can probably avoid this function altogether.